### PR TITLE
Adding Application and supportive files(README, Marathon JSONs, Jenkinsfile, Dockerfile) 

### PR DESCRIPTION
**Pull request by sergiimatusEPAM**

### CODE REVIEW

**Comment by sergiimatusEPAM** [comment]
@jeschkies , @DominikDary 
please review README.md, Marathon JSONs, Jenkinsfile, Dockerfile

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+def DockerRegToken = "DockerHub_token"
+def DockerfilePath = "Dockerfile"
+def DockerAppImageName = "${dockerHubAccount}/${appName}"
+
+def buildNumber = "${BUILD_NUMBER}"
+def jobName = "${JOB_NAME}"
+////////////////////////
+
+node("${dynamicJenkinsSlave}") {
```
Comment (jeschkies) [inline Jenkinsfile]: Any reason this is not simply "windows"?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    stage("Checkout of git repo (cmd)") {
+        bat "IF not exist ${appName} (git clone ${gitURL}) else (cd ${appName} && git pull)";
```
Comment (jeschkies) [inline Jenkinsfile]: I think the Jenkins checkout should just do that.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    stage("Clean") {
+        bat "cd ${appName} && dotnet clean"
+    }
+    stage("Build") {
+        bat "cd ${appName} && dotnet build"
+    }
+    stage("Publish, pack into zip") {
```
Comment (jeschkies) [inline Jenkinsfile]: Could this be done by the build tool?

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        bat "cd ${appName} && dotnet publish -o .\\..\\..\\target"
+        bat "7z.exe a -mmt2 -tzip package.${buildNumber}.zip target"
+		bat "copy /b/v/y package.${buildNumber}.zip demoapp.zip"
+    }
+    stage("Build Docker image"){
+        bat """
+            docker build -t ${DockerAppImageName} ${DockerfilePath} --no-cache
```
Comment (jeschkies) [inline Jenkinsfile]: Can the Docker build build the app instead?

File: Marathon-templates/configuration.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
+jenkins:
```
Comment (jeschkies) [inline Marathon-templates/configuration.yaml]: This should not be part of this diff IMHO.

File: Jenkinsfile (hunk 1 above)
Comment (sergiimatusEPAM) [reply Jenkinsfile]: I've modified the pipeline and made as scripted just to predict case when an user will have few windows labels: windows1, windows2 , etc

File: Jenkinsfile (hunk 3 above)
Comment (sergiimatusEPAM) [reply Jenkinsfile]: it's not a maven and we have no single artifact pre-pared. Therefore packing this into zip

Comment (jeschkies) [reply Jenkinsfile]: Sure but even `make` would be better than having build steps in the `Jenkinsfile`.

File: Jenkinsfile (hunk 1 above)
Comment (jeschkies) [reply Jenkinsfile]: I don't think that's a requirement. Users should create new Jenkinsfiles for each project and change the label there.

File: Jenkinsfile (hunk 4 above)
Comment (sergiimatusEPAM) [reply Jenkinsfile]: a build of the app happens in a docker container, and then it's 2nd stage - where shipping of a zip file happens into runtime docker container.

File: Marathon-templates/configuration.yaml (hunk 5 above)
Comment (sergiimatusEPAM) [reply Marathon-templates/configuration.yaml]: excluding

**Comment by sergiimatusEPAM** [comment]
@jongiddy , please review as well

**Comment by sergiimatusEPAM** [comment]
@jongiddy , please review

**Comment by sergiimatusEPAM** [comment]
@DominikDary ,  adding you to review process. Welcome back! )

File: Jenkinsfile (hunk 4 above)
Comment (jeschkies) [reply Jenkinsfile]: I'm not sure I understand. I mean that `dotnet build` should happen during `docker build` IMHO. See the Jenkins Mesos plugin [Dockerfile](https://github.com/mesosphere/jenkins-mesos-plugin/blob/master/Dockerfile#L5) as an example. The build can be a stage in the Dockerfile and not in the Jenkinsfile.

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            docker tag ${DockerAppImageName}:latest ${DockerAppImageName}:0.${buildNumber}
+        """
+    }
+    stage("Publish to DockerHub"){
+        withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: "${DockerRegToken}", usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD']]) {
+            bat """
+                echo | set /p="%PASSWORD%" | docker login -u %USERNAME% --password-stdin
```
Comment (jeschkies) [inline Jenkinsfile]: Please use `withDockerRegistry` instead.

File: Marathon-templates/jenkins.json [hunk 7]
```diff
@@ -0,0 +1,129 @@
+{
```
Comment (jeschkies) [inline Marathon-templates/jenkins.json]: I think this can go.

File: dotnet-builder/Dockerfile.jenkins.windows.slave [hunk 8]
```diff
@@ -0,0 +1,14 @@
+FROM winamd64/openjdk:8-jre-windowsservercore-1809 AS builder
```
Comment (jeschkies) [inline dotnet-builder/Dockerfile.jenkins.windows.slave]: This should go against https://github.com/mesosphere/dcos-jenkins-dind-agent

**Pull request closed without merging**